app/src/main/java/com/example/munchkin/Karte/HandKarten.java

```diff
@@ -71,17 +73,41 @@ void onKartenSlotClicked(int i)
     //Added eine neue Karte an eine freie Handkarten Position
     public void addKarte(Karte karte)
     {
-        if(checkIfNotMoreThan4())
+        if(countHandkarten() < 7)
             getNextEmptySlot().karteAblegen(karte);
     }
 
     public  void addKarte(Karte[] karten)
     {
         for(int i = 0; i < karten.length; i++)
         {
-            if(checkIfNotMoreThan4())
-            {
+            if(countHandkarten() < 7)
                 getNextEmptySlot().karteAblegen(karten[i]);
+        }
+    }
+
+    //TODO: Should be called when the player finishes his round
+    public void onFinishRound() {
+        // Only up to 5 Cards are allowed
+        if(countHandkarten() > 5) {
+            int playerLevel = Player.getLocalPlayer().getPlayerLevel().getLevel();
+            Player lowestPlayer = Player.getLocalPlayer();
+            int currentLowest = playerLevel;
+            for(int i = 0; i < Lobby.getPlayers().length; ++i) {
+                if(Lobby.getPlayers()[i] != null && Lobby.getPlayers()[i].getPlayerLevel().getLevel() < currentLowest) {
+                    lowestPlayer = Lobby.getPlayers()[i];
+                    currentLowest = Lobby.getPlayers()[i].getPlayerLevel().getLevel();
+                }
+            }
+
+            for(int i = 5; i < kartenSlots.length; ++i) {
+                if(lowestPlayer.equals(Player.getLocalPlayer())) {
+                    // We are the lowest player so move card to ablage
+                    this.removeKarte(i);
+                } else {
+                    // Give card to player with lowest level
+                    lowestPlayer.getInventar().getHandKarten().addKarte(this.removeKarte(i));
```

Comment:
lowestPlayer.getInventar().getHandKarten().addKarte(this.removeKarte(i)); Funktioniert nicht, da das Inventar von den anderen Spielern nicht per Networking synchronisiert wird. Muss mittels Networking gemacht werden. Sonst super implementiert :)